pkg/stencil/lockfile.go

```diff
@@ -81,3 +84,70 @@ func LoadLockfile(path string) (*Lockfile, error) {
 	err = yaml.NewDecoder(f).Decode(&lock)
 	return lock, err
 }
+
+// MergeMissingFilesFromOlderLockfile merges missing files from the older lockfile into the current one, for use with file.Once.
+func (lf *Lockfile) MergeMissingFilesFromOlderLockfile(older *Lockfile) {
+	for _, f := range older.Files {
+		if !slices.ContainsFunc(lf.Files, func(fe *LockfileFileEntry) bool {
+			return fe.Name == f.Name
+		}) {
+			lf.Files = append(lf.Files, f)
+		}
+	}
+
+	lf.Sort()
+}
+
+// Sort maintains the alphabetic sort of files/modules to ensure deterministic output
+func (lf *Lockfile) Sort() {
+	sort.SliceStable(lf.Files, func(i, j int) bool {
+		return lf.Files[i].Name < lf.Files[j].Name
+	})
+
+	sort.SliceStable(lf.Modules, func(i, j int) bool {
+		return lf.Modules[i].Name < lf.Modules[j].Name
+	})
+}
+
+// Write writes the finished lockfile out to disk
+func (lf *Lockfile) Write() error {
+	f, err := os.Create(LockfileName)
+	if err != nil {
+		return fmt.Errorf("failed to create lockfile: %w", err)
+	}
+	defer f.Close()
+
+	enc := yaml.NewEncoder(f)
+	defer enc.Close()
+
+	if err := enc.Encode(lf); err != nil {
+		return fmt.Errorf("failed to write lockfile: %w", err)
+	}
+
+	return nil
+}
+
+func (lf *Lockfile) Prune() []string {
```

Comment:
I think it'd be a good idea to plumb through two things to prune:
1. A slice of modules
2. A slice of file paths

If both are empty it prunes everything as you already have it here. If one or both have entries they only prune in accordance to what was passed (by module(s) and/or file path(s))

Reply:
Modules aren't being left in the lockfile, only files. I was thinking about allowing a specific set of files to prune, so maybe i'll just do that.

Reply:
I wonder if we should leave module, though. It'd help track down what version of a module added a file that's left in the lockfile.

Reply:
shrug.  i don't see it being useful, but it's easy enough to leave in.